modules/core/02-client/keeper/keeper.go

```diff
@@ -322,6 +323,36 @@ func (k *Keeper) GetSelfConsensusState(ctx sdk.Context, height exported.Height)
 	return k.consensusHost.GetSelfConsensusState(ctx, height)
 }
 
+// SetCounterparty sets the Counterparty for a given client identifier.
+func (k *Keeper) SetCounterparty(ctx sdk.Context, clientID, counterpartyClientID string, merklePathPrefix *commitmenttypes.MerklePath) {
+	counterparty := types.NewCounterparty(counterpartyClientID, merklePathPrefix)
+	bz := k.cdc.MustMarshal(&counterparty)
+	k.ClientStore(ctx, clientID).Set(host.CounterpartyKey(), bz)
+}
+
+// GetCounterparty gets the counterparty client's identifier for a given client identifier.
+func (k *Keeper) GetCounterparty(ctx sdk.Context, clientID string) (types.Counterparty, bool) {
+	store := k.ClientStore(ctx, clientID)
+	bz := store.Get(host.CounterpartyKey())
+	if len(bz) == 0 {
+		return types.Counterparty{}, false
+	}
+
+	var counterparty types.Counterparty
+	k.cdc.MustUnmarshal(bz, &counterparty)
+	return counterparty, true
+}
+
+// GetCreator returns the creator of the client.
+func (k *Keeper) GetCreator(ctx sdk.Context, clientID string) string {
+	return string(k.ClientStore(ctx, clientID).Get([]byte(types.CreatorKey)))
```

Comment:
I feel like we should keep with the consistency of ibc-go Get/Set functions? That is, this function should return a bool if found or not?

Reply:
Happy to go here with the consistent approach if we have consensus, is it fine for everyone?